src/nvim/mapping.c

```diff
@@ -2396,10 +2407,16 @@ void modify_keymap(uint64_t channel_id, Buffer buffer, bool is_unmap, String mod
     KEY_TO_BOOL(script);
     KEY_TO_BOOL(expr);
     KEY_TO_BOOL(unique);
+    KEY_TO_BOOL(replace_keycodes);
 #undef KEY_TO_BOOL
   }
   parsed_args.buffer = !global;
 
+  if (parsed_args.replace_keycodes && !parsed_args.expr) {
+    api_set_error(err, kErrorTypeValidation,  "\"replace_keycodes\" requires \"expr\"");
```

Comment:
Maybe replace_keycodes without expr should be ignored?

Reply:
Error is usually a good idea, espcially if it can be done trivially as it is here. The exception is when there's a common expected use-case where it's convenient for callers.

Reply:
`desc` without callback is not ignored.

Reply:
@zeertzjq It is for me? `:lua vim.api.nvim_set_keymap('n', '<cr>', '<cr>', { desc = '...' })`

Reply:
I mean maybe not ignored, but it's useless without the callback, isn't it?

Reply:
Why useless? It is shown in `:map`.

Reply:
I didn't notice that :) My bad then, I thought it was only supposed to describe lua functions, because you couldn't print the function source like you can with vim script.